### ISSUE TITLE
MixtureArray - equivalent for SolutionArray

### ISSUE THREAD

**Issue by ischoegl** [Cantera/enhancements#73]
**Abstract**

Multiple states of a single `Solution` phase can be conveniently processed using a `SolutionArray`. At the moment, there is no equivalent for `Mixture` objects, which complicates saving and reimporting of solutions.

**Motivation**

Describe the need for the proposed change:

* What problem is it trying to solve? ... facilitate work with solutions
* Who is affected by the change? ... users working with simulations involving multiple states of `Mixture` objects
* Why is this a good solution? ... manual conversion to `mix.T, mix.P, mix.species_moles` works, but is not convenient

**Possible Solutions**

A `MixtureArray` class can likely be constructed in an analogous way to `SolutionArray`

For file IO, multiple realizations are possible:

1. underlying `SolutionArray` objects for saving/loading (easy to implement)
2. use `Mixture.species_moles` for saving/loading (more convenient for use with 3rd party post-processing tools)

**Comment by speth** [comment]
I think this is a good idea. The one bit of caution I would add is that implementing this will almost certainly drag up a lot of the deficiencies of the `Mixture` (aka C++ `MultiPhase` class). That class appears to have originally been created only for the purpose of defining the input conditions for a multi-phase equilibrium calculation, and as a result it's missing a lot of methods that you'd expect for inspecting and modifying the state.

**Comment by ischoegl** [comment]
I agree that this may dredge up too many other deficiencies for `Mixture`s, which may need to be addressed in order to make this useful. (e.g. #64 has higher priority).
